hack/getting_started.sh

```diff
@@ -0,0 +1,551 @@
+    else
+        grep -q "byoh" ${clusterCtlYamlFile}
+        if [ $? -ne 0 ] ; then
+            writeByoh=1
+        fi
+    fi 
+
+    if [ ${writeByoh} -eq 1 ] ; then
+        cat>>${clusterCtlYamlFile}<<EOF
+providers:
+  - name: byoh
+    url: https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/latest/infrastructure-components.yaml
+    type: InfrastructureProvider
+EOF
+    fi
+
+    clusterctl config repositories | grep -q byoh
+    if [ $? -ne 0 ] ; then
+        echo "Config clusterctl failed..."
+        exit
+    fi
+}
+
+function runByohAgent(){
+    index=$1
+    byohBinaryFile=$2
+    manageClusterConfFile=$3
+
+    runCmd "docker cp ${byohBinaryFile} host${index}:/byoh-hostagent" 0 "Copying agent binary to byoh container: host${index}..."
+    runCmd "docker cp ${manageClusterConfFile} host${index}:/management-cluster.conf" 0 "Copying kubeconfig to byoh container: host${index}..."
+    docker exec -d host$i sh -c "chmod +x /byoh-hostagent && /byoh-hostagent --kubeconfig management-cluster.conf --skip-installation > /agent.log 2>&1"
+    if [ $? -ne 0 ] ; then
+        echo "Starting the host${index} agent..."
+        exit
+    fi
+
+    
+    maxRunTimes=10
+    waitTime=1
+    for((i=1;i<=${maxRunTimes};i++));  
+    do   
+        kubectl get byohosts host${index} | grep -v NAME | grep -q host${index}
+        if [ $? -ne 0 ]; then
+            sleep ${waitTime}
+            continue
+        else   
+            echo "byohost object(host${index}) is created successfully..."
+            return
+        fi
+    done
+
+    echo "Error: byohost object(host${index}) is created failed..."
+    exit
+}
+
+function installDocker() {
+    cmdName="docker"
+
+    ## check  if dependency is present
+    isCmdInstalled "${cmdName}"
+
+    if [ $? -ne 0 ]; then
+        return
+    fi
+
+    ## install it if it not installed
+    echo "Installing docker...."
+    runCmd "sudo apt update" 0
+    runCmd "sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common" 0
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu  $(lsb_release -cs)  stable"
+    runCmd "sudo apt-get install -y docker-ce" 0
+
+    ## check  if denpency is installed successfully
+    isCmdInstalled  "${cmdName}"
+    exitIfNot $? 0 "Installing ${cmdName} failed, exit..."
+    runCmd "sudo systemctl enable docker" 0
+}
+
+function enableDocker() {
+    runCmd "sudo systemctl start docker" 0
+    runCmd "sudo systemctl enable docker" 0
+
+    sudo systemctl status docker | grep -q "active (running)"
+    if [ $? -ne 0 ]; then
+        echo "Enable docker failed"
+        exit 
+    fi
+    echo "Enable docker success"
+
+    # Make sure current user has permission for docker, do this if not Create the docker group.
+    docker ps 2>&1 | grep -q "connect: permission denied"
```

Comment:
I think we should just exit with the error. This is making the script complicated and a getting started script should not be handling creating users / groups etc.